db/schema.sql

```diff
@@ -74,7 +74,7 @@ CREATE INDEX client_reports_task_and_time_index ON client_reports(task_id, nonce
 -- Specifies the possible state of an aggregation job.
 CREATE TYPE AGGREGATION_JOB_STATE AS ENUM(
     'IN_PROGRESS', -- at least one included report is in a non-terminal (START, WAITING) state
-    'FINISHED'     -- all reports have reached a terminal state (FINISHED, FAILED, INVALID)
+    'FINISHED'     -- all reports have reached a terminal state (FINISHED, FAILED, INVALID) or we have given up on the aggregation job entirely
```

Comment:
Would it be better to create a new state for aggregation jobs that exhaust retries? (maybe 'ABANDONED'?) This would make identifying affected jobs easier, and would simplify the correspondence between report aggregation states and aggregation job states.

Reply:
I like this idea; I was trying to decide whether to do something like this myself. While I think it's not strictly necessary technically speaking, a separate state will certainly make analysis easier (as you note). And I suppose that a separate state may make later changes less error-prone -- while we don't currently assume that an aggregation job in state "FINISHED" has all-terminal report aggregations today, it'd be easy to assume so later since that would be _almost_ true.

I went ahead and made this change.